### ISSUE TITLE
Sorted sets: score precision loss

### ISSUE THREAD

**Issue by vapongit** [phpredis/phpredis#1865]
The results of ZADD command execution through redis-cli and phpredis are different.
Example:
- redis-cli:
```
127.0.0.1:6379> ZADD SET_NAME 12199283400115995 KEY_NAME
(integer) 1
127.0.0.1:6379> ZSCAN SET_NAME 0
1) "0"
2) 1) "KEY_NAME"
   2) "12199283400115996"

```
- php code:
```php
$redis = new Redis();
$redis->connect('127.0.0.1', 6379);
$redis->zadd('SET_NAME', 12199283400115995, 'KEY_NAME');
$cursor = null;
$data = $redis->zScan('SET_NAME', $cursor,null,1000);
var_export($data);

```
output:
```php
array (
  'KEY_NAME' => 12199283400116000.0,
)
```
Also redis-cli monitor shows a command generated by phpredis: 
`"ZADD" "SET_NAME" "1.2199283400116e+16" "KEY_NAME"`

A possible cause of precision loss:
- redis code uses "%.17g" sprintf format for doubles, see [src/networking.c:607](https://github.com/redis/redis/blob/bea40e6a41e31a52e9e6efee77ea5a4bd873b759/src/networking.c#L607)
- phpredis code uses "%.16g" sprintf format for doubles, see [library.c:954](https://github.com/phpredis/phpredis/blob/81c502ae7c0de65d63cd514ee59849c9d1b0b952/library.c#L954)

**Comment by michael-grunder** [comment]
Thanks for the report.

It does look like the issue has to do with `%.16g` vs `%.17g`.

I'm certain we picked `%.16g` because that's what Redis was using, but I'm only seeing `%.17g` in Redis' git history.  :shrug:

**Comment by yatsukhnenko** [comment]
Fixed in 5.3.2
